backend/gs/file_test.go

```diff
@@ -113,6 +114,118 @@ func (ts *fileTestSuite) TestRead() {
 	ts.Equal(localFile.String(), contents, "Copying an gs file to a buffer should fill buffer with file's contents")
 }
 
+func (ts *fileTestSuite) TestDelete() {
+	contents := "hello world!"
+	bucketName := "bucki"
+	objectName := "some/path/file.txt"
+	server := fakestorage.NewServer(
+		Objects{
+			fakestorage.Object{
+				ObjectAttrs: fakestorage.ObjectAttrs{
+					BucketName:      bucketName,
+					Name:            objectName,
+					ContentType:     "text/plain",
+					ContentEncoding: "utf8",
+				},
+				Content: []byte(contents),
+			},
+		},
+	)
+	defer server.Stop()
+	client := server.Client()
+	fs := NewFileSystem().WithClient(client)
+
+	file, err := fs.NewFile(bucketName, "/"+objectName)
+	if err != nil {
+		ts.Fail("Shouldn't fail creating new file")
+	}
+
+	err = file.Delete()
+	if err != nil {
+		ts.Fail("Shouldn't fail deleting the file")
+	}
+
+	bucket := client.Bucket(bucketName)
+	assert.Equal(ts.T(), false, objectExists(bucket, objectName))
+}
+
+func (ts *fileTestSuite) TestDeleteError() {
+	contents := "hello world!"
+	bucketName := "bucki"
+	objectName := "some/path/file.txt"
+	server := fakestorage.NewServer(
+		Objects{
+			fakestorage.Object{
+				ObjectAttrs: fakestorage.ObjectAttrs{
+					BucketName:      bucketName,
+					Name:            objectName,
+					ContentType:     "text/plain",
+					ContentEncoding: "utf8",
+				},
+				Content: []byte(contents),
+			},
+		},
+	)
+	defer server.Stop()
+	client := server.Client()
+	fs := NewFileSystem().WithClient(client)
+
+	file, err := fs.NewFile(bucketName, "/invalidObject")
+	if err != nil {
+		ts.Fail("Shouldn't fail creating new file")
+	}
+
+	err = file.Delete()
+	ts.NotNil(err, "Should return an error if gs client had error")
+}
+
+func (ts *fileTestSuite) TestDeleteRemoveAllVersions() {
```

Comment:
For this test, I couldn't use more than one versions because in the fakestorage when you add second object with same name, the first object is moved under archived objects list https://github.com/fsouza/fake-gcs-server/blob/main/internal/backend/memory.go#L43 and when you do deletes on all versions,[ it looks up object using just the bucket name and object name]( https://github.com/fsouza/fake-gcs-server/blob/main/fakestorage/object.go#L517) and doesn't use generation id so it [defaults to using generation '0'](https://github.com/fsouza/fake-gcs-server/blob/665c3a68049611ce8d0006faaf94c4414fe87396/internal/backend/memory.go#L227). When generation 0 is used, it always does a lookup only in active objects so during first call it is able to find the object in activeObjects list but when it tries to find the object with different generation the next time since the generation is still set to 0, it errors out saying object not found.